macro-utils/src/tuple.rs

```diff
@@ -60,7 +60,14 @@ fn display(ast: &syn::DeriveInput) -> syn::Result<TokenStream> {
         },
         Type::Vec(_, subtype) => match Type::from_path(&subtype) {
             Type::String(_) | Type::Simple(_) | Type::Struct(_) => quote! {
-                write!(f, "{}", self.0.iter().join(" "))
+                if let Some(fst) = self.0.first() {
+                    write!(f, "{}", fst)?;
+                    for elem in &self.0[1..] {
+                        write!(f, " {}", elem)?;
+                    }
+                }
```

Comment:
A bit more readable version:

```suggestion
                let mut it = self.0.iter();

                if let Some(val) = it.next() {
                    write!(f, "{}", val)?;
                }

                while let Some(val) = it.next() {
                    write!(f, " {}", val)?;
                }
```